zrml/swaps/src/lib.rs

```diff
@@ -1068,6 +1084,10 @@ mod pallet {
         ValueQuery,
     >;
 
+    #[pallet::storage]
+    #[pallet::getter(fn pools_cached_for_arbitrage)]
+    pub type PoolsCachedForArbitrage<T: Config> = StorageMap<_, Twox64Concat, PoolId, ()>;
```

Comment:
I think it's kind of an edge case. Yes, you have control over the key. But you can't just push arbitrary keys, because you can only push those keys for which pools actually exist.

Reply:
d693094c57aceaa2d6857e7e0028220df33b44e3

Reply:
OK Malte, good point, yes only pools that are already existing are allowed. @sea212  Would `Twox64Concat` allow an attack vector for this case ?

Reply:
No, attackers have no influence on the sequence of `PoolId`s.

Reply:
Ok, then we should use `Twox64Concat`.

Reply:
fe36c492460d00a94dd9c5d05e3790afc875ece4